### PR TITLE
Bug 1735141: Fixes conccurency issue with multiple-thread trying to trigger already running jobs

### PR DESCRIPTION
**Pull request by akram**
Fixes conccurency issue with multiple-thread trying to trigger already running jobs

### CODE REVIEW

**Comment by gabemontero** [comment]
Fix the PR title .... "Bug 1735141: < ... some text summarizing the change ...>"

**Comment by gabemontero** [comment]
just saw the wip label ... yeah @akram just do an assign to me when you think this is ready for me to look at re: our discussion in slack just now

**Comment by akram** [comment]
I will just force commit a smaller version for review, and I will keep the other refactors on a second branch for later merge.

**Comment by akram** [comment]
/assign @gabemontero 

@gabemontero could you PTAL ?

**Comment by akram** [comment]
@gabemontero I am facing an unexpected behaviour while doing additional tests: if I run `oc start-build` my jenkins restarts

**Comment by akram** [comment]
> @gabemontero I am facing an unexpected behaviour while doing additional tests: if I run `oc start-build` my jenkins restarts

ok, looks like it was a resource issue on my server, I got OOMKilled

File: src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java [hunk 1]
```diff
@@ -375,65 +358,64 @@ else if (envKeys.contains(param.getName())) {
...
 	}
 
-    public static boolean triggerJob(WorkflowJob job, Build build)
-            throws IOException {
-        if (isAlreadyTriggered(job, build)) {
-            return false;
-        }
-
+	  /**
+     * @param job   to trigger
```
Comment (andyfeller) [inline src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java]:
```suggestion
     * @param job to trigger
```

**Comment by akram** [comment]
Hi @andyfeller , thank you for your review. I just pushed the last commit instead, and will provide a larger refactor PR that I saved locally including your remarks.

**Pull request merged**